### PR TITLE
Fix muted and joined channel list queries with empty data (Auto Filtering Enabled)

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/423

### 🎯 Goal
Fix the channel list query being empty when the `.muted` and `.joined` filters were used with auto filtering enabled.

### 🛠 Implementation
We currently have an issue with our Auto Filtering where the `.muted` and `.joined` filters do not properly map to our DB predicates.

After some analysis and discussion with @ipavlidakis, we opted for introducing a `PredicateMapper` so that a filter can override the default Predicate logic.

### 🧪 Manual Testing Notes
Unit tests were added to prove it works correctly.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChat/Query/Filter.swift [hunk 1]
```diff
@@ -111,13 +111,19 @@ public struct Filter<Scope: FilterScope> {
     /// Whether the `keyPathString` represents an array in the DTO entities.
     let isCollectionFilter: Bool
 
+    /// The mapper that will override the DB Predicate. This might be needed
```
Comment (ipavlidakis) [inline Sources/StreamChat/Query/Filter.swift:114]: The comment is very helpful. Thanks for adding it!

**Review by ipavlidakis — APPROVED** [review]
LGTM 🚀

File: Sources/StreamChat/Query/ChannelListQuery.swift [hunk 2]
```diff
@@ -113,12 +113,36 @@ public extension FilterKey where Scope: AnyChannelListFilterScope {
...
+    /// Supported operators: `equal`.
+    static var muted: FilterKey<Scope, Bool> { .init(
+        rawValue: "muted",
+        keyPathString: #keyPath(ChannelDTO.mute),
+        predicateMapper: { op, muted in
+            let key = #keyPath(ChannelDTO.mute)
+            switch op {
+            case .equal:
+                return NSPredicate(format: muted ? "\(key) != nil" : "\(key) == nil")
+            default:
```
Comment (polqf) [inline Sources/StreamChat/Query/ChannelListQuery.swift:141]: The code looks the same in both implementations (but the keyPath). Should we be reusing it by placing it in a single place? We are probably using the same logic in other places?

**Review by polqf — APPROVED** [review]

File: Sources/StreamChat/Query/ChannelListQuery.swift (hunk 2 above)
Comment (nuno-vieira) [reply Sources/StreamChat/Query/ChannelListQuery.swift:141]: For now, I think it is fine to have a little duplication. We can introduce a helper to reuse this logic if we see more use cases like this. 👍

**Pull request merged**